docs/extend/server-schemas.md

```diff
@@ -0,0 +1,64 @@
+# Server component schemas
+
+Starting with Zowe v2.0, each Component in Zowe must contain a [json schema](https://json-schema.org/) describing the configuration parameters that are valid for its component section in Zowe's server configuration. If a component does not have anything that can be configured, this file can just be boilerplate specifying that it fully inherits generic Component parameters and nothing more.
+
+The server infrastructure will utilize each components' schema files to validate a Zowe instance configuration every startup, so this requirement is enforced by code.
+
+## Requirements
+
+* Server component json schema files must follow the json schema [spec 2019-09](https://json-schema.org/draft/2019-09/schema).
+* Each component must state where its base schema file is located by the manifest parameter "schemas.config"
+* The schema file must use and/or extend the Zowe Component base schema by use of the "allOf" attribute.
+* The schema must have an `$id` property which is a URI that has a domain related to the entity that developed the Component.
+* The file should be tagged on z/OS but elsewhere must at least be encoded as ASCII-subset of UTF-8
+
+## Additional information
+
+* The schema file can reference other schema files within the component if compartmentalization of definitions are desired
+
+
+## Example
+
+Below is an example manifest and schema for a Component named "component1". The manifest file specifies the location of the schema file, and the schema file specifies the configuration parameters that are valid for this Component.
+
+### Example manifest
+
+```
+name: component1
+id: com.abcdef.component1
+title: component1
+description: An example component with a simple configuration.
+license: ABCDEF Company License
+schemas:
+  - config: schema.json
+```
+
+
+### Example schema
+
+Below is an example of the "schema.json" file referenced above. In it, we have 1 special property, "my-custom-prop", which is just a boolean that can be true or false.
+
+```
+{
+  "$schema": "https://json-schema.org/draft/2019-09",
+  "$id": "https://abcdef.com/schema/component1",
+  "allOf": [
+    { "$ref": "https://zowe.org/schema/base" },
+    { "$ref": "https://zowe.org/schema/base#zoweComponent"},
```

Comment:
Should we call them `https://zowe.org/schemas/server-component-base` to distinguish with CLI config schema. And with plural `schemas`.

Reply:
How versioning could be affect this url? I'm not sure what's the best practice here.

Reply:
sounds good on name changed.
not sure about versions, checking on this...
i know our schemas will be versioned by github tag but we may need to incorporate version into the URL as json-schema.org does as well.
and uh, these URLs dont need to be real URLs but perhaps we could publish ours there for convenience.

Reply:
checked and its suggested that semver be in the URL

Reply:
i have a concern about the version number in the schema id
basically we should be able to add stuff every zowe release and we can diff to see
but because component schemas need to ref ours, if we change the version number in the id, it will break.
instead we could be a little nonstandard by making a version property separately.